assets/js/view/buy_trains.rb

```diff
@@ -95,7 +95,7 @@ def from_depot(depot_trains)
 
     def other_trains(other_corp_trains)
       other_corp_trains.flat_map do |other, trains|
-        trains.map do |train|
+        trains.uniq(&:name).map do |train|
```

Comment:
how about 

trains.group_by(&:name).map do |train, trains|

"Train #{train.name} ... #{trains.size}"

Reply:
Got close with the latest commit